server/repository/src/db_diesel/changelog/README.md

```diff
@@ -0,0 +1,46 @@
+# Changelog
+
+A number of operations rely on processing a `queue` of database changes.
+These operations include pushing records to central server during synchronisation, processing transfer records, processing messaging, aggregation, interoperability synchronisation, etc.. 
+
+Generic mechanism for recording and querying these database changes was implemented in the form of the `changelog` table.
+
+## Triggers
+
+Database triggers insert new entries into changelog table with:
+
+* cursor <- sequential number for order of database operation
+* record_id 
+* table_name
+* store_id <- the store that `owns` the record
+* name_id <- the name_id of the `destination` entity (can also be a store)
```

Comment:
side thought: would it be more helpful if the field was `destination_name_id`? it would therefore be self-documenting.

Reply:
hmm, i like this but only if all other instances have the same, if it's name_id in requisition and invoice but it's different in changelog i think it becomes inconsistent